### PR TITLE
Add e2e test for prometheus monitoring

### PR DESCRIPTION
**Pull request by avorima**
## Description

Add basic e2e test for prometheus monitoring feature that checks that resources are deployed and cleaned up successfully.
Depends on #7425 to fix the schema defaulting

Also introduce a general k8sobject e2e test helper package in favor of specialized helpers.

ICP-9590

## How can this be tested?

```
make test/e2e/prometheus-monitoring
```

### CODE REVIEW

**Review by aorcholski — DISMISSED** [review]
LGTM

File: test/e2e/features/prometheusmonitoring/prometheusmonitoring.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+	"sigs.k8s.io/e2e-framework/third_party/helm"
+)
+
+const (
+	defaultTargetAllocatorRepo       = "public.ecr.aws/dynatrace/otel-target-allocator"
+	targetAllocatorImageEnvVar       = "E2E_TARGET_ALLOCATOR_IMAGE"
+	targetAllocatorImageDigestEnvVar = "E2E_TARGET_ALLOCATOR_IMAGE_DIGEST"
+)
+
+func Feature(t *testing.T) features.Feature {
```
Comment (andriisoldatenko) [inline test/e2e/features/prometheusmonitoring/prometheusmonitoring.go:33]: can we write second test with all public registry images?

**Comment by andriisoldatenko** [comment]
code LGTM

**Comment by andriisoldatenko** [comment]
```
Failed
=== RUN   TestNoCSI_prometheus_monitoring/lifecycle
I0915 18:25:23.382328   16426 helm.go:253] "Determining if helm binary is available or not" executable="helm"
I0915 18:25:23.382447   16426 helm.go:262] "Running Helm Operation" command="helm upgrade dynatrace-operator --namespace dynatrace --create-namespace --install --rollback-on-failure --set platform=openshift --set installCRD=true --set csidriver.enabled=false --set manifests=true --set debugLogs=true --set image=ghcr.io/dynatrace/dynatrace-operator:snapshot-prometheus-e2e --set imageRef.pullPolicy=Always /home/runner/_work/dynatrace-operator/dynatrace-operator/target/config/helm/chart/default --set experimental.enablePrometheus=true --set prometheus.installCRDs=true --kubeconfig ''"
I0915 18:25:26.159913   16426 helm.go:269] Helm Command output 
exit status 1
    feature.go:34: 
        	Error Trace:	/home/runner/_work/dynatrace-operator/dynatrace-operator/target/test/e2e/helpers/feature.go:34
        	            				/home/runner/go/pkg/mod/sigs.k8s.io/e2e-framework@v0.7.0/pkg/env/env.go:479
        	            				/home/runner/go/pkg/mod/sigs.k8s.io/e2e-framework@v0.7.0/pkg/env/env.go:493
        	            				/home/runner/go/pkg/mod/sigs.k8s.io/e2e-framework@v0.7.0/pkg/env/env.go:237
        	Error:      	Received unexpected error:
        	            	Error: UPGRADE FAILED: unable to continue with update: CustomResourceDefinition "podmonitors.monitoring.coreos.com" in namespace "" exists and cannot be imported into the current release: invalid ownership metadata; label validation error: key "app.kubernetes.io/managed-by" must equal "Helm": current value is "cluster-version-operator"; annotation validation error: missing key "meta.helm.sh/release-name": must be set to "dynatrace-operator"; annotation validation error: missing key "meta.helm.sh/release-namespace": must be set to "dynatrace": exit status 1
        	Test:       	TestNoCSI_prometheus_monitoring/lifecycle
I0915 18:25:26.160097   16426 events.go:22] "test failed" f="TestNoCSI_prometheus_monitoring" failed=true
I0915 18:25:26.181185   16426 events.go:33] "Events list" events total=0
    --- FAIL: TestNoCSI_prometheus_monitoring/lifecycle (2.80s)
```

@avorima please take a look, e2e test on CI failed.

File: test/e2e/features/prometheusmonitoring/prometheusmonitoring.go (hunk 1 above)
Comment (avorima) [reply test/e2e/features/prometheusmonitoring/prometheusmonitoring.go:33]: it doesn't work on my env yet, so i can't verify it

**Comment by avorima** [comment]
> @avorima please take a look, e2e test on CI failed.

it looks like they have prometheus operator installed. it's one of the edge cases we should document

File: test/e2e/features/prometheusmonitoring/prometheusmonitoring.go (hunk 1 above)
Comment (andriisoldatenko) [reply test/e2e/features/prometheusmonitoring/prometheusmonitoring.go:33]: we decided to cover it in another story/epic.

**Review by andriisoldatenko — DISMISSED** [review]

**Review by aorcholski — DISMISSED** [review]

File: test/e2e/features/prometheusmonitoring/prometheusmonitoring.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+	"sigs.k8s.io/e2e-framework/third_party/helm"
+)
+
+const (
+	defaultTargetAllocatorRepo       = "public.ecr.aws/dynatrace/otel-target-allocator"
+	targetAllocatorImageEnvVar       = "E2E_TARGET_ALLOCATOR_IMAGE"
+	targetAllocatorImageDigestEnvVar = "E2E_TARGET_ALLOCATOR_IMAGE_DIGEST"
```
Comment (mihaitanasedt) [inline test/e2e/features/prometheusmonitoring/prometheusmonitoring.go:30]: This looks like dead code  `targetAllocatorImageDigestEnvVar `

Comment (avorima) [reply test/e2e/features/prometheusmonitoring/prometheusmonitoring.go:30]: i added a test case to use the variable

**Review by mihaitanasedt — APPROVED** [review]
